sovrin_client/agent/endpoint.py

```diff
@@ -11,27 +11,40 @@
 logger = getlogger()
 
 
-class Endpoint(SimpleStack):
+class Endpoint:
```

Comment:
well, most of your comments are valid, and I am not sure if Jason will approve this, and we were aware about few of the things like pass through methods etc. 

We wanted to run through this way to see if things looks good, but again, I believe both approach has some pros and cons. Even with this PR, I wasn't able to solve the core concern raise by Jason about initializing objects in start method etc. 

The core thing is that we wanted to defer the port binding when agent starts not when it gets created, the problem is creation of SimpleStack class object simply binds the port that time. So, unless we wants to go ahead and change code in raet, we have limited options (let it be how originally it was, we know it might be wrong to bind port during object creation **OR** shift SimpleStack initialiation during start method or something).